pandas/core/arrays/interval.py

```diff
@@ -934,9 +934,7 @@ def copy(self: IntervalArrayT) -> IntervalArrayT:
     def isna(self) -> np.ndarray:
         return isna(self._left)
 
-    def shift(
-        self: IntervalArrayT, periods: int = 1, fill_value: object = None
-    ) -> IntervalArray:
+    def shift(self, periods: int = 1, fill_value: object = None) -> IntervalArray:
```

Comment:
why are these not valid?

Reply:
same here:
`IntervalArrayT = TypeVar("IntervalArrayT", bound="IntervalArray")`
if the return value was of type `IntervalArrayT`, it would be valid.